circuit_maintenance_parser/providers.py

```diff
@@ -73,12 +76,15 @@ def process(self, data_type: str = "") -> Iterable[Maintenance]:
                 )
                 if data_type and data_type == parser.get_data_type() or not data_type:
                     return parser.process()
-            except (ParsingError, MissingMandatoryFields):
+            except (ParsingError, MissingMandatoryFields) as exc:
                 logger.debug(
-                    "Parser %s for provider %s was not successful", parser_class.__name__, self.__class__.__name__
+                    "Parser %s for provider %s was not successful: %s", parser_name, provider_name, exc,
                 )
+                error_message += f"- Parser class {parser_name} from {provider_name} failed due: {exc}\n"
                 continue
-        raise ParsingError("None of the parsers was able to parse the notification")
+        raise ParsingError(
+            f"None of the {provider_name} parsers was able to parse the notification.\nDetails:\n{error_message}"
+        )
```

Comment:
Would it be reasonable to additionally attach the list of actual exceptions as an additional attribute to the ParsingError, so that the caller has the option to introspect them if desired?